crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -914,15 +914,13 @@ unsafe impl<'__w, T: Component> WorldQuery for &'__w mut T {
             StorageType::Table => {
                 let (table_components, added_ticks, changed_ticks) =
                     fetch.table_data.debug_checked_unwrap();
-                Mut {
-                    value: table_components.get(table_row.index()).deref_mut(),
-                    ticks: TicksMut {
-                        added: added_ticks.get(table_row.index()).deref_mut(),
-                        changed: changed_ticks.get(table_row.index()).deref_mut(),
-                        change_tick: fetch.change_tick,
-                        last_change_tick: fetch.last_change_tick,
-                    },
-                }
+                Mut::new(
```

Comment:
IMO this is harder to read than the raw construction. It also makes it a lot easier to mess up without tighter typing on each of the arguments. I'd love to remove the `pub(crate)` designations on each of the fields, but IMO this can be more readily written.

Reply:
So do you just want me to revert this one use of `Mut::new`?

Reply:
No I'd rather all of the internal uses of it be reverted. Or alternatively we should properly newtype everything to potentially avoid botching its use internally.